traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices_required_capabilities.go

```diff
@@ -219,22 +219,51 @@ func (rc *RequiredCapability) getCapabilities(tenantIDs []int) ([]tc.DeliverySer
 // Delete implements the api.CRUDer interface.
 func (rc *RequiredCapability) Delete() (error, error, int) {
 	authorized, err := rc.isTenantAuthorized()
-	if err != nil {
-		return nil, errors.New("checking tenant: " + err.Error()), http.StatusInternalServerError
-	} else if !authorized {
+	if !authorized {
 		return errors.New("not authorized on this tenant"), nil, http.StatusForbidden
+	} else if err != nil {
+		return nil, fmt.Errorf("checking authorization for existing DS ID: %s" + err.Error()), http.StatusInternalServerError
+	}
+
+	// Check if the Delivery Service is associated with the Required Capability
+	statusCode, err := rc.exists()
+	if err != nil {
+		return err, nil, statusCode
 	}
 
 	return api.GenericDelete(rc)
 }
 
+// exists checks to see if a required capability is assigned to a delivery service
+func (rc *RequiredCapability) exists() (int, error) {
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(rc.APIInfo().Params, rc.ParamColumns())
+	if len(errs) > 0 {
+		return http.StatusBadRequest, util.JoinErrs(errs)
+	}
+	fmt.Printf("\n\nwhere: %+v\norderby: %+v\npag: %+v\nqueryvalues: %+v\n\n", where, orderBy, pagination, queryValues)
+
+	var count int
+	query := `SELECT count(deliveryservice_id) FROM deliveryservices_required_capability where deliveryservice_id = $1 and required_capability = $2`
```

Comment:
nothing to address but instead of COUNT you could simplify this to SELECT EXISTS and read that into a bool